### PR TITLE
add a test for makedumpfile merge/split

### PR DESCRIPTION
**Pull request by emmawu**
Signed-off-by: xiawu <xiawu@redhat.com>

### CODE REVIEW

File: makedumpfile-merge-split/runtest.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+DUMPLV=${DUMPLV:="31"}
+DUMPFILE=${DUMPFILE:=${K_TMP_DIR}/vmcore.$(date +%s)}
+MISC=${MISC:=""}
+CMPRSS=${CMPRSS:=""}
+EPPIC=${EPPIC:=""}
+CONFIG=${CONFIG:="scrub.conf"}
+VMLINUX="/usr/lib/debug/lib/modules/$(uname -r)/vmlinux"
+
+
+VerifyScrub()
```
Comment (linuxqiao) [inline makedumpfile-merge-split/runtest.sh]: We should keep the consistent format, use 'verify_scrub' as a function name.

File: makedumpfile-merge-split/runtest.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        if [[ "${line}" =~ (jiffies|utime)\ =\ .*0x([[:xdigit:]]+) && \
+          ! "${BASH_REMATCH[2]}" =~ ^(58)+$ ]]; then
+            return 1
+        fi
+    done < "${log}"
+    return 0
+}
+
+
+ParseOption()
```
Comment (linuxqiao) [inline makedumpfile-merge-split/runtest.sh]: same as line:32

File: makedumpfile-merge-split/runtest.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    [ "${retval}" -ne 0 ] && log_error "- The makedumpfile command returns ${retval}"
+
+    log_info "- Validating the merged dumpfile."
+    VerifyScrub
+
+    ready_to_exit
+}
+
+log_info "- Start"
+makedumpfile-merge-split
```
Comment (linuxqiao) [inline makedumpfile-merge-split/runtest.sh]: same as line:32

File: makedumpfile-merge-split/scrub.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
+string
```
Comment (linuxqiao) [inline makedumpfile-merge-split/scrub.c]: In this c file, i think we can use "<type> <function-name>" as a c file format.

File: lib/crash.sh [hunk 5]
```diff
@@ -44,25 +44,21 @@ get_vmcore_path()
...
+        local export_path
+        export_path=$(cat "${K_NFS}")
         vmcore_path=${export_path}${vmcore_path}
     fi
 
     local vmcore_full_path
     vmcore_full_path=$(find "${vmcore_path}" -newer /etc/kdump.conf -name "${vmcore_name}" -type f)
-    count=$(echo $vmcore_full_path | wc -w)
-    if [ $count -gt 1 ]; then
+    count=$(echo "${vmcore_full_path}" | wc -w)
```
Comment (sztsian) [inline lib/crash.sh]: Suggest not to use 'wc -w'.

File: makedumpfile-merge-split/scrub.c [hunk 6]
```diff
@@ -0,0 +1,38 @@
+string scrub_opt()
```
Comment (sztsian) [inline makedumpfile-merge-split/scrub.c]: General question, do we know who created this .c file, or what's the license for this?

File: makedumpfile-merge-split/runtest.sh [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author: Guangze Bai <gbai@redhat.com>
```
Comment (sztsian) [inline makedumpfile-merge-split/runtest.sh]: If it is not from existing file, I prefer to omit gbai@. If it is, then it's fine.

**Comment by linuxqiao** [comment]
I don't know, maybe it's no any license. I just know it is create by gbai@.

--
Thanks,
Qiao

File: makedumpfile-merge-split/runtest.sh (hunk 3 above)
Comment (emmawu) [inline makedumpfile-merge-split/runtest.sh]: Thanks. Updated the names of all 3 functions.

File: makedumpfile-merge-split/scrub.c (hunk 4 above)
Comment (emmawu) [inline makedumpfile-merge-split/scrub.c]: Updated the c script. Moved the function name up to the same line as the return type.

File: lib/crash.sh (hunk 5 above)
Comment (emmawu) [inline lib/crash.sh]: Updated script to get $count number.

**Comment by emmawu** [comment]
@sztsian, @linuxqiao 
Done updating the code based on your comments. Pls review it again.

**Pull request merged**